app/src/main/java/com/example/eric/mdbweather/MainActivity.java

```diff
@@ -0,0 +1,314 @@
+
+                                JSONArray minutely_array = minutely_weather.getJSONArray(getString(R.string.json_minutely_data));
+                                int minutely_array_len = minutely_array.length();
+                                for(int i = 0; i < minutely_array_len; i++){
+                                    JSONObject minute = minutely_array.getJSONObject(i);
+                                    double rain_minute = minute.getDouble(getString(R.string.json_minutely_array_precipProb));
+                                    if(rain_minute > 0){
+                                        String rain_time = Utils.epocToDateWithTime(minute.getLong(getString(R.string.json_minutely_array_time)));
+                                        String precip_type = minute.getString(getString(R.string.json_minutely_array_precipType));
+                                        current_rain.setText("Incoming " + precip_type + " around " + rain_time);
+                                        break;
+                                    } else {
+                                        current_rain.setText(getString(R.string.no_rain));
+                                    }
+                                }
+
+                                String date = Utils.epocToDate(time);
+
+                                current_date.setText(date);
+                                current_temp.setText(Integer.toString(temp) + (char)0x00B0 + " F");
+                                current_summary.setText(summary.substring(0, summary.length() - 1));
+
+                            } catch (Exception e) {
+                                e.printStackTrace();
+                            }
+                        }
+                    }.execute();
+                    return rootView;
+
+                case 2:
+                    rootView = inflater.inflate(R.layout.fragment_hours, container, false);
+                    RecyclerView recyclerView2 = (RecyclerView) rootView.findViewById(R.id.recyclerView2);
+                    ListAdapterHours adapter2 = new ListAdapterHours(latitude, longitude, getString(R.string.darksky_api_key));
+                    recyclerView2.setLayoutManager(new LinearLayoutManager(getActivity()));
+                    recyclerView2.setAdapter(adapter2);
+                    return rootView;
+
+                case 3:
+                    rootView = inflater.inflate(R.layout.fragment_days, container, false);
+                    RecyclerView recyclerView = (RecyclerView) rootView.findViewById(R.id.recyclerView);
+                    ListAdapter adapter = new ListAdapter(latitude, longitude, getString(R.string.darksky_api_key));
+                    recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
+                    recyclerView.setAdapter(adapter);
+                    return rootView;
+            }
+            return null;
```

Comment:
You're essentially using this as a default value in the switch case, better explicitly put it in the default block